cmake/module/Maintenance.cmake

```diff
@@ -88,3 +88,64 @@ function(add_windows_deploy_target)
     add_custom_target(deploy DEPENDS ${CMAKE_BINARY_DIR}/bitcoin-win64-setup.exe)
   endif()
 endfunction()
+
+function(add_macos_deploy_target)
+  if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND TARGET bitcoin-qt)
+    set(macos_app "Bitcoin-Qt.app")
+    # Populate Contents subdirectory.
+    configure_file(${PROJECT_SOURCE_DIR}/share/qt/Info.plist.in ${macos_app}/Contents/Info.plist)
+    file(CONFIGURE OUTPUT ${macos_app}/Contents/PkgInfo CONTENT "APPL????")
+    # Populate Contents/Resources subdirectory.
+    file(CONFIGURE OUTPUT ${macos_app}/Contents/Resources/empty.lproj CONTENT "")
+    configure_file(${PROJECT_SOURCE_DIR}/src/qt/res/icons/bitcoin.icns ${macos_app}/Contents/Resources/bitcoin.icns COPYONLY)
+    file(CONFIGURE OUTPUT ${macos_app}/Contents/Resources/Base.lproj/InfoPlist.strings
+      CONTENT "{ CFBundleDisplayName = \"@PACKAGE_NAME@\"; CFBundleName = \"@PACKAGE_NAME@\"; }"
+    )
```

Comment:
This appears to run as part of the configure step and so we have an incomplete app bundle called Bitcoin-Qt.app created in the build directory.

Should these steps be part of their own custom command and only called as part of the deploydir and deploy targets? And perhaps removed after the zip is created?

Reply:
> Should these steps be part of their own custom command and only called as part of the deploydir and deploy targets? 

I've considered this option while working on this PR and chosen this implementation for the following reasons:
- the content of all files above is known at the configuration stage
- it is simpler / more readable 

I'd be happy to consider benefits of the alternative approach, if any.

> And perhaps the file cleaned up after the zip is created?

Why?

Reply:
> Why?

As a noob and after running the build I see this bundle in the build root and think it's _the_ app bundle and not just a template used to produce a real one in `dist`.

It's only a NIT and just something unobvious to me so I had to investigate what it was.